plugins/module_utils/template.py

```diff
@@ -429,3 +429,30 @@ def update_config_with_template_and_references(self, config_data, reference_coll
     def check_template_when_name_is_provided(self, parameter):
         if parameter and parameter.get("name") and not (parameter.get("template") or parameter.get("template_id")):
             self.mso.fail_json(msg="Either 'template' or 'template_id' associated with '{}' must be provided".format(parameter.get("name")))
+
+
+class MSOTemplateCache:
+    def __init__(self):
+        self.cache = {}
+        self.template_property_cache = {}
```

Comment:
not sure I understand why this is needed?

Reply:
Done!